src/plugins/controls/public/control_group/editor/edit_control_button.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { EuiButtonIcon } from '@elastic/eui';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import UseUnmount from 'react-use/lib/useUnmount';
+
+import { ControlGroupInput } from '../types';
+import { OverlayRef } from '../../../../../core/public';
+import { ControlGroupStrings } from '../control_group_strings';
+import { controlGroupReducers } from '../state/control_group_reducers';
+import { useReduxContainerContext } from '../../../../presentation_util/public';
+import { openEditControlFlyout } from './open_edit_control_flyout';
+
+export const EditControlButton = ({ embeddableId }: { embeddableId: string }) => {
+  // Redux embeddable container Context
+  const reduxContainerContext = useReduxContainerContext<
+    ControlGroupInput,
+    typeof controlGroupReducers
+  >();
+  const {
+    containerActions: { untilEmbeddableLoaded, removeEmbeddable, updateInputForChild },
+    actions: { setControlWidth },
+    useEmbeddableSelector,
+    useEmbeddableDispatch,
+  } = reduxContainerContext;
+  const dispatch = useEmbeddableDispatch();
+
+  // current state
+  const { panels } = useEmbeddableSelector((state) => state);
+  // keep up to date ref of latest panel state for comparison when closing editor.
+  const latestPanelState = useRef(panels[embeddableId]);
+  useEffect(() => {
+    latestPanelState.current = panels[embeddableId];
+  }, [panels, embeddableId]);
+
+  const panel = panels[embeddableId];
+
+  const [currentFlyout, setCurrentFlyout] = useState<OverlayRef | undefined>(undefined);
+
+  const closeEditFlyout = useCallback(() => {
+    if (currentFlyout) currentFlyout.close();
+    setCurrentFlyout(undefined);
+  }, [currentFlyout]);
+
+  const openEditFlyout = async () => {
+    await untilEmbeddableLoaded(embeddableId).then((embeddable) => {
+      setCurrentFlyout(
+        openEditControlFlyout({
+          embeddable,
+          latestPanelState: latestPanelState.current,
+          panel,
+          updateInputForChild,
+          removeEmbeddable,
+          updateWidth: (newWidth) => dispatch(setControlWidth({ width: newWidth, embeddableId })),
+          closeFlyout: () => closeEditFlyout(),
+        })
+      );
+    });
+  };
+
+  UseUnmount(() => {
+    closeEditFlyout();
```

Comment:
As a bonus, this component is unmounted when the user switches to view mode - so, without any extra work, `UseUnmount` will make the flyout close for both (1) navigating away from the dashboard and (2) switching to view mode 👍 Cool!